lib/galaxy/tools/toolbox/watcher.py

```diff
@@ -15,7 +15,13 @@
     can_watch = False
 
 from galaxy.util.hash_util import md5_hash_file
-from galaxy.web.stack import register_postfork_function
+try:
+    from galaxy.web.stack import register_postfork_function
+except ImportError:
+
+    # noop, replacement for non-web environments
+    def register_postfork_function(f):
+        pass
```

Comment:
Shouldn't this match the signature of `register_postfork_function`, which is `f, *args, **kwargs`. Next question, shouldn't we immediately execute the function ?